src/Executor/ReferenceExecutor.php

```diff
@@ -1393,4 +1417,18 @@ protected function ensureValidRuntimeType(
 
         return $runtimeType;
     }
+
+    /**
+     * @param mixed $contextValue
+     *
+     * @return mixed
+     */
+    private function maybeScopeContext($contextValue)
+    {
+        if ($contextValue instanceof ScopedContext) {
+            return clone $contextValue;
```

Comment:
```suggestion
            return $contextValue->clone();
```

This let's the interface be more than a marker and have more control, otherwise the implementation is also trivial.